cardano-node/src/Cardano/Node/TUI/LiveView.hs

```diff
@@ -572,32 +576,64 @@ initLiveViewState = do
-  modifyMVar_ (getbe lvbe) $ \lvs ->
-    return lvs { lvsNodeId = pack $ "Port: " <> show (naPort nodeAddr) }
-setTopology lvbe npm@NodeCLI{nodeMode = MockProtocolMode} = do
-  nc <- parseNodeConfiguration npm
-  modifyMVar_ (getbe lvbe) $ \lvs ->
-    return $ lvs { lvsNodeId = namenum (ncNodeId nc) }
+
+-- | Change a few fields in the LiveViewState after it has been initialized above.
+liveViewPostSetup :: NFData a => LiveViewBackend blk a -> NodeCLI -> NodeConfiguration-> IO ()
+liveViewPostSetup lvbe ncli nc = do
+    modifyMVar_ (getbe lvbe) $ \lvs ->
+      pure lvs
+            { lvsNodeId = nodeId
+            , lvsProtocol = ncProtocol nc
+            , lvsRelease = if ncProtocol nc == TPraos then "Shelley" else "Byron"
+
+            -- Hard code these here, but they get properly updated in setNodeKernel
+            -- about 30 seconds after the node starts.
+            , lvsMempoolCapacity = 2000
+            , lvsMempoolCapacityBytes = 4096000
+
+
+            }
  where
-  namenum (Just (CoreId (CoreNodeId num))) = "C" <> pack (show num)
-  namenum (Just (RelayId num)) = "R" <> pack (show num)
-  namenum Nothing = panic $ "Cardano.Node.TUI.LiveView.namenum: "
-                          <> "Mock protocols require a NodeId value in the configuration .yaml file"
+    nodeId :: Text
+    nodeId =
+      case nodeMode ncli of
+        MockProtocolMode -> namenum (ncNodeId nc)
+        RealProtocolMode -> Text.pack $ "Port: " <> show (naPort $ nodeAddr ncli)
+
+    namenum :: Maybe NodeId -> Text
+    namenum mnid =
+      case mnid of
+        Just (CoreId (CoreNodeId num))-> "C" <> Text.pack (show num)
+        Just (RelayId num) -> "R" <> Text.pack (show num)
+        Nothing ->
+          panic $ "Cardano.Node.TUI.LiveView.namenum: "
+                <> "Mock protocols require a NodeId value in the configuration .yaml file"
+
+
 
 setNodeThread :: NFData a => LiveViewBackend blk a -> Async.Async () -> IO ()
 setNodeThread lvbe nodeThr =
   modifyMVar_ (getbe lvbe) $ \lvs ->
       return $ lvs { lvsNodeThread = LiveViewThread $ Just nodeThr }
 
-setNodeKernel :: NFData a
+setNodeKernel :: (NFData a, HasTxMaxSize (ExtLedgerState blk))
               => LiveViewBackend blk a
               -> NodeKernel IO RemoteConnectionId LocalConnectionId blk
               -> IO ()
 setNodeKernel lvbe nodeKern =
-    modifyMVar_ (getbe lvbe) $ \lvs ->
-        return $ lvs { lvsNodeKernel = SJust (LVNodeKernel nodeKern) }
+    modifyMVar_ (getbe lvbe) $ \lvs -> do
+        -- This is correct for Byron and Shelley.
+        MempoolCapacityBytes mempoolCapacity <- STM.atomically $ getCapacity (getMempool nodeKern)
+
+        currentLedger <- STM.atomically $ getCurrentLedger (getChainDB nodeKern)
+
+        pure lvs
+              { lvsNodeKernel = SJust (LVNodeKernel nodeKern)
+              , lvsMempoolCapacity =
+                  fromIntegral $
+                    mempoolCapacity `div` getMaxTxSize currentLedger
```

Comment:
So this is a conservative estimate, I see.

Reply:
Yes, I understand. This is how the code esitmated the capacity based on hard coded values before.